r2-core/src/main/java/com/linkedin/r2/filter/transport/ClientRetryFilter.java

```diff
@@ -73,8 +74,13 @@ public void onStreamError(Throwable ex,
     {
       nextFilter.onError(new RetriableRequestException(retryAttr), requestContext, wireAttrs);
     }
-    else
-    {
+    else {
+      Throwable[] throwables = ExceptionUtils.getThrowables(ex);
+      for (Throwable throwable : throwables) {
+        if (throwable instanceof RetriableRequestException) {
+          ((RetriableRequestException) throwable).setDoNotRetryOverride(true);
```

Comment:
Break this loop if one throwable is retriable?

Reply:
Fixed